dm/master/config.go

```diff
@@ -439,17 +439,21 @@ func parseURLs(s string) ([]url.URL, error) {
 	return urls, nil
 }
 
-func genEmbedEtcdConfigWithLogger() *embed.Config {
+func genEmbedEtcdConfigWithLogger(logLevel string) *embed.Config {
 	cfg := embed.NewConfig()
 	cfg.EnableGRPCGateway = true // enable gRPC gateway for the internal etcd.
 
 	// use zap as the logger for embed etcd
 	// NOTE: `genEmbedEtcdConfig` can only be called after logger initialized.
 	// NOTE: if using zap logger for etcd, must build it before any concurrent gRPC calls,
 	// otherwise, DATA RACE occur in NewZapCoreLoggerBuilder and gRPC.
-	// NOTE: we can only increase the log level for the clone logger but not decrease.
-	logger := log.L().WithFields(zap.String("component", "embed etcd")).WithOptions(zap.IncreaseLevel(zap.ErrorLevel))
-	cfg.ZapLoggerBuilder = embed.NewZapCoreLoggerBuilder(logger, logger.Core(), log.Props().Syncer) // use global app props.
+	logger := log.L().WithFields(zap.String("component", "embed etcd"))
+	// if logLevel is info, set etcd log level to WARN to reduce log
+	if strings.ToLower(logLevel) == "info" {
+		logger.Logger = logger.WithOptions(zap.IncreaseLevel(zap.WarnLevel))
+	}
+
+	cfg.ZapLoggerBuilder = embed.NewZapCoreLoggerBuilder(logger.Logger, logger.Core(), log.Props().Syncer) // use global app props.
```

Comment:
not sure why change `logger` to `logger.Logger`

Reply:
type of logger now is log.Logger after https://github.com/pingcap/dm/blob/7de7bb4fcbe6085cb8f5d7a8fd50589465e7b918/dm/master/config.go#L451 change. The type should be log.Logger.logger